src/ring_theory/dedekind_domain.lean

```diff
@@ -945,4 +945,124 @@ begin
+ (hn : (unique_factorization_monoid.factors I).count p ≤ n):
+  (map I^.quotient.mk p)^n = (map I^.quotient.mk p)^((factors I).count p) :=
+begin
+  rw [← map_pow, ← map_pow],
+  have H : map I^.quotient.mk ((p^n)) = map I^.quotient.mk (p^n ⊔ I),
+  { rw [map_sup, map_mk_eq_bot_of_le (le_refl I), sup_bot_eq] },
+  rw sup_dvd_of_ge I hI p Hp n hn at H,
+  exact H,
+end
+
+lemma pow_map_gt_of_exponent_lt (hI : I ≠ ⊥) (p : ideal T) (Hp : p ∈ factors I) (n : ℕ)
+  (hn : n < (factors I).count p) :
+  (map I^.quotient.mk p)^(factors I).count p < (map I^.quotient.mk p)^n :=
+begin
+  suffices H : p^(factors I).count p < p^n,
+  { have H₁ : (map I^.quotient.mk p)^(factors I).count p = (map I^.quotient.mk p)^n *
+    (map I^.quotient.mk p)^((factors I).count p - n),
+    { by rw [← pow_add, nat.add_sub_cancel' (le_of_lt hn)] },
+    apply lt_of_le_of_ne (le_of_dvd ⟨(map I^.quotient.mk p)^((factors I).count p - n), H₁⟩),
+    by_contra hcontra,
+    { rw [ne.def, not_not] at hcontra,
+      apply_fun (comap I^.quotient.mk) at hcontra,
+      rw [← map_pow, ← map_pow, comap_map_of_surjective I^.quotient.mk quotient.mk_surjective,
+        comap_map_of_surjective I^.quotient.mk quotient.mk_surjective, ← ring_hom.ker_eq_comap_bot,
+        mk_ker, sup_dvd_of_ge I hI p Hp ((factors I).count p) (le_refl ((factors I).count p)),
+        sup_dvd_of_lt I hI Hp n hn] at hcontra,
+      rw hcontra at H,
+      exact false_of_ne (ne_of_lt H) } },
+  have H' : p^(factors I).count p = p^n * p^((factors I).count p - n) :=
+    by rw [← pow_add, nat.add_sub_cancel' (le_of_lt hn)],
+  rw ← ideal.dvd_not_unit_iff_lt,
+  split,
+  { exact pow_ne_zero n (prime_of_factor p Hp).ne_zero },
+  { exact ⟨p^((factors I).count p - n),
+      not_is_unit_of_not_is_unit_dvd (prime_of_factor p Hp).not_unit
+      (dvd_pow (dvd_refl p) (ne_of_gt (sub_pos_iff_lt.mpr hn))), H'⟩ }
+end
+
+lemma shifted_seq_pow_constant_iff_ge_count (hI : I ≠ ⊥) (p: ideal T) (hp : p ∈ factors I) (n : ℕ):
+  shifted_seq_pow_constant I p n ↔ (factors I).count p ≤ n :=
+begin
+  split,
+  { intro H,
+    by_contradiction hc,
+    have H₄ := pow_map_gt_of_exponent_lt I hI p hp n (not_le.1 hc),
+    rw (H ((factors I).count p) (le_of_lt (not_le.1 hc))).symm at H₄,
+    apply false_of_ne,
+    exact ne_of_lt H₄ },
+  { rw shifted_seq_pow_constant,
+    intro hn,
+    intros m hm,
+    rw [seq_pow_eventually_constant I hI p hp n hn,
+    seq_pow_eventually_constant I hI p hp m (le_trans hn hm)] }
+end
+
+lemma seq_pow_eventually_constant' (hI : I ≠ ⊥) (p : ideal T) (hp : p ∈ factors I) :
+  shifted_seq_pow_constant I p ((factors I).count p) :=
+(shifted_seq_pow_constant_iff_ge_count I hI p hp ((factors I).count p)).2
+  (le_refl ((factors I).count p))
+
+
+/--The quotient multiplicity of `p` is the least natural number `n` such that the sequence of
+ powers `(p')^m` for `m ≥ n` is constant, where `p'` is the image of `p` in `R/I` -/
```

Comment:
This docstring needs some work. The ideal `I` should be mentioned earlier. It is the *quotient multiplicity* of `p` relative to the ideal `I`, right? Something like that.

Do you have a reference for this concept? It is the first time I have heard of it.

Reply:
Sure, I'll write something something more precise.

I don't think this concept ever get a name in the literature, since it's just another way of expressing the multiplicity of `p` in the factorization of `I` (quotient multiplicity is just an ad hoc name I came up for this)